src/clnt_vc.c

```diff
@@ -312,7 +312,7 @@ clnt_vc_ncreate2(int fd,	/* open file descriptor */
 		if (ct->ct_addr.len)
 			mem_free(ct->ct_addr.buf, ct->ct_addr.len);
 
-		if (xd->refcnt == 1) {
```

Comment:
I agree that checking for 1 looks like the correct thing to do.  If it was created, it was initialized to 1 and never modified; if it was not created, it was incremented.  In either case, it cannot be 0, so this change at least introduces dead code, and likely a leak as well.  A quick scan indicates it should be >1 if it is reused, so the check for 1 looks correct.